src/rank_llm/retrieve/retriever.py

```diff
@@ -182,6 +224,27 @@ def retrieve(self) -> List[Dict[str, Any]]:
             retrieved_results = [
                 Result(query=r["query"], hits=r["hits"]) for r in self._dataset
             ]
+        elif self._retrieval_mode == RetrievalMode.CUSTOM:
+            candidates_file = Path(
+                f"retrieve_results/{self._retrieval_method.name}/retrieve_results_{self._dataset}.json"

```

Comment:
This is still using retrieval method and dataset which won't be populated in this code path, please use the following arguements for candidate file name: index_path=index_path,
topics_path=topics_path,
index_type=index_type,
encoder=encoder,
onnx=onnx,

Reply:
this has been done already, I set the `self._dataset` to be the index directory [here](https://github.com/jasper-xian/rank_llm/blob/4a7146be7f595231678e1df3d46224e383887c1a/src/rank_llm/retrieve/retriever.py#L172)

retrieval_method I set to be `UNSPECIFIED`, and I think that works fine when giving a custom index

so basically, the candidate filepath would be `retrieve_results/UNSPECIFIED/retrieve_results_{index_dir}` where `index_dir` = last part of the index filepath

let me know if this works for you, or what changes you want to make to this

Reply:
hm, but this means same index path with different topics_path will have false collision.
In the case of retrieving from the dataset, all the input params are in the file name, it means the file names will match only when all specified params are identical. I think it should be the same for the custom index path, to avoid false collisions.
I also think instead of UNSPECIDIED, it makes sense to have "CUSTOM_INDEX" as a second level dir for custom index path.

Reply:
I see, okay I'll address this

Reply:
changes made @sahel-sh , new filename is [here](https://github.com/jasper-xian/rank_llm/blob/fe7e0f01860f0dfcccf1622a6f59535957bed25e/src/rank_llm/retrieve/retriever.py#L170), let me know if that works or if it is too long